### PR TITLE
My Receipts Page

### LINKED ISSUE #19: My Receipts Page UI
My Receipts Page UI from Figma Designs

Image

### LINKED ISSUE #36: Receipts Page (Connect to Backend)

### PR DESCRIPTION
**Pull request by SamuelLiSiyuan**
My Receipts Page UI and logic,

- My benefit on bottom bar is now changed to my receipts page
- Dropdown filter by used cards and purchase type
- Addtion infomation field for each receipt
- Record receipt button at the bottom (not functional)

myReceiptsPage

### CODE REVIEW

**Comment by SamuelLiSiyuan** [comment]
What should be on the View Receipt Page?

**Review by NandiniMeh — COMMENTED** [review]
On the View Receipt page, it should just be the receipt itself, and the information related to it (like date, purchase type, amount, card used etc) not sure if anyone else is working on this though @rachellefontanilla @sophiec103 @jeanniezhang

This is regarding the button for "view" right

**Comment by SamuelLiSiyuan** [comment]
> date, purchase type, amount, card used

Since all this information is already shown on this page, if no more detailed information needs to be presented, I am not sure if it is necessary to have a separate page for it. 

Anyway, if no one is working on the view receipt page, I can take it.

**Comment by NandiniMeh** [comment]
> > date, purchase type, amount, card used
> 
> Since all this information is already shown on this page, if no more detailed information needs to be presented, I am not sure if it is necessary to have a separate page for it.
> 
> Anyway, if no one is working on the view receipt page, I can take it.

hmmm thats a good point. maybe a way to see the image would be good? any suggestions would be great. like for example, if there's a way to get the image of the receipt to be saved to gallery or something

File: app/src/main/java/edu/card/clarity/domain/creditCard/CreditCardInfo.kt
Comment (LideLinusZhang) [inline app/src/main/java/edu/card/clarity/domain/creditCard/CreditCardInfo.kt:1]: Don't pollute the domain model with ui state stuff such as `@Parcelize`.

**Review by LideLinusZhang — CHANGES_REQUESTED** [review]

File: app/src/main/java/edu/card/clarity/presentation/myReceiptsScreen/MyReceiptsScreenViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                    null, "All", RewardType.CashBack,
+                    CardNetworkType.MasterCard,
+                    Calendar.getInstance(),
+                    Calendar.getInstance(), false
+                )
+            )
+        )
+    }
+
+    /*// Temporary Data
```
Comment (LideLinusZhang) [inline app/src/main/java/edu/card/clarity/presentation/myReceiptsScreen/MyReceiptsScreenViewModel.kt]: Please remove the commented code.

**Review by LideLinusZhang — APPROVED** [review]

**Pull request merged**